js/js_test.go

```diff
@@ -167,3 +168,57 @@ func TestType(t *testing.T) {
 		t.Errorf("got %s, want %s", got, want)
 	}
 }
+
+func TestValueOf(t *testing.T) {
+	for _, test := range []struct {
+		in         interface{}
+		wantType   js.Type
+		wantString string
+	}{
+		{js.Value(js.ValueOf(42)), js.TypeNumber, "42"},
+		{js.NewCallback(func(args []js.Value) {}), js.TypeFunction, ""},
+		{js.TypedArrayOf([]int8{1, 2, 3}), js.TypeObject, "1,2,3"},
+		{nil, js.TypeNull, "null"},
+		{bool(true), js.TypeBoolean, "true"},
+		{int(1), js.TypeNumber, "1"},
+		{int8(2), js.TypeNumber, "2"},
+		{int16(3), js.TypeNumber, "3"},
+		{int32(4), js.TypeNumber, "4"},
+		{int64(5), js.TypeNumber, "5"},
+		{uint(6), js.TypeNumber, "6"},
+		{uint8(7), js.TypeNumber, "7"},
+		{uint16(8), js.TypeNumber, "8"},
+		{uint32(9), js.TypeNumber, "9"},
+		{uint64(10), js.TypeNumber, "10"},
+		{float32(11), js.TypeNumber, "11"},
+		{float64(12), js.TypeNumber, "12"},
+		// FIXME this doesn't work {unsafe.Pointer(&x19), js.TypeNumber, ""},
+		{string("hello"), js.TypeString, "hello"},
+		{map[string]interface{}{"a": 1, "b": "c"}, js.TypeObject, "[object Object]"},
```

Comment:
Would it be possible to get better strings by `JSON.stringify`?

Reply:
Have done that and pushed an updated version